docs/reactivity.md

```diff
@@ -0,0 +1,410 @@
+
+```js echo
+{
+  const z = 3;
+}
+```
+
+If multiple blocks define top-level variables with the same name, these blocks will still run, but any references to duplicated variables in other blocks will throw a duplicate definition error because the definition is ambiguous.
+
+## Promises
+
+<div class="tip">See <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises">MDN’s <i>Using promises</i></a> for an introduction to promises.</div>
+
+A [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) represents the result of an asynchronous operation: some data loaded from a file, say, or a module imported dynamically, or the end of an animation, or some text input from the user.
+
+In Framework, when one code block refers to a promise defined in another code block, the referencing code block implicitly awaits the promise. This means promises are often invisible — you don’t have to worry whether something is a promise or whether it’s resolved.
+
+<div class="note">Implicit <code>await</code> only applies <i>across</i> code blocks, not <i>within</i> a code block. Within a code block, a promise is just a promise.</div>
+
+For example, below `FileAttachment.json` returns a promise, and so the value of `volcano` is a promise.
+
+```js echo
+const volcano = FileAttachment("volcano.json").json();
+```
+
+And yet if we reference `volcano` in another code block or inline expression, we don’t need to `await`. The `await` is implicit; the code block automatically waits for the `volcano` promise to resolve before running.
+
+```js echo
+volcano
+```
+
+The volcano dataset has ${volcano.values.length.toLocaleString("en-US")} values.
+
+```md run=false
+The volcano dataset has ${volcano.values.length.toLocaleString("en-US")} values.
+```
+
+This pattern is especially useful for loading multiple files. The files are loaded in parallel, and referencing code blocks only wait for the files they need. This is faster than loading files sequentially with `await`, and simpler than `Promise.all`.
+
+```js run=false
+const a = FileAttachment("a.csv").csv({typed: true});
+const b = FileAttachment("b.csv").csv({typed: true});
+const c = FileAttachment("c.csv").csv({typed: true});
+```
+
+Implicit await causes the entire code block to wait, not just expressions that reference promises. This distinction is usually invisible, but you might notice if you have a code block that references both “slow” and “fast” promises: the code block waits for all promises to resolve, and thus is gated by the slowest promise. Below, “fast” and “slow” are printed at the same time.
+
+```js echo
+const fast = new Promise((resolve) => setTimeout(() => resolve("fast"), 500));
+const slow = new Promise((resolve) => setTimeout(() => resolve("slow"), 5000));
+```
+```js echo
+display(fast);
+display(slow);
+```
```

Comment:
No, that would negate the purpose of this example, which is to distinguish between expression-level implicit await (a hypothesized misunderstanding of how implicit await could be interpreted) and code block-level implicit await. 😅 

Imagine that implicit `await` happened at the expression level. Then the code would implicitly be like this:

```js
display(await slow);
display(await fast);
```

If we took your order, then there is no difference between expression level await and block level await: the fast promise would already be resolved after the slow promise resolved, and so they would print at the same time.

The point of this example is to show that the behavior in Observable is different than this, because the await happens at the block level. So now imagine that we had expression level implicit await in Observable. This code would behave differently than what you see in Observable:

```js
display(await fast);
display(await slow);
```

Even though the fast promise resolves quickly, you still have to await for the slow promise before either displays. Block level await means that the code is like this:

```js
Promise.all([fast, slow]).then(([fast, slow]) => {
  display(fast);
  display(slow);
});
```